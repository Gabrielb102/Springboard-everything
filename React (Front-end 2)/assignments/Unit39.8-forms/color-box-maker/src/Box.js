import "./Box.css";

const Box = ({color, width, height, removeFunc}) => {
    const style = {backgroundColor: color, height: height, width: width}; 
    return (
    <div className="Box" data-testid="box">
        <div style={style}></div>
        <button data-testid="x" onClick={removeFunc}>x</button>
    </div>
)}

export default Box;