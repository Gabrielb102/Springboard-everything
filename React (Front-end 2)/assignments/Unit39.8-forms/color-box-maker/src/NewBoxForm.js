import {useState} from "react";
import Box from "./Box";


const NewBoxForm = ({addBoxFunc}) => {
    const INITIALCONDITIONS = {
        color : '',
        height : '',
        width : ''
    }

    const [formData, setFormData] = useState(INITIALCONDITIONS);

    const handleChange = (e) => {
        const {name, value} = e.target;
        setFormData(formData => ({
            ...formData,
            [name] : value
        }))
    };
    
    const handleSubmit = (e) => {
        e.preventDefault()
        const {color, height, width} = formData;
        const cssHeight = height + "px";
        const cssWidth = width + "px";
        addBoxFunc(color, cssHeight, cssWidth);
        setFormData(INITIALCONDITIONS);
    }

    return (
        <form data-testid="form" onSubmit={handleSubmit}>
            <label htmlFor="color">Color</label>
            <input id="color" name="color" value={formData.color} onChange={handleChange} placeholder="css colors only please"></input>            
            <label htmlFor="height">Height</label>
            <input id="height" name="height" value={formData.height} onChange={handleChange}></input>            
            <label htmlFor="width">Width</label>
            <input id="width" name="width" value={formData.width} onChange={handleChange}></input>            
            <button type="submit">New Box</button>
        </form>
    )
}

export default NewBoxForm;