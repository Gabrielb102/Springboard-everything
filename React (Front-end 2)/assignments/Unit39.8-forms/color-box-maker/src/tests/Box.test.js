import { render, fireEvent, queryByTestId, getByTestId } from "@testing-library/react";
import Box from "../Box";
import NewBoxForm from "../NewBoxForm";

describe("Tests for Box Component", () => {

    it("Renders", () => {
        render(<Box/>);
    })

    it("Renders a consistent user interface", () => {
        const id = '3516516548'
        const { asFragment } = render(<Box id={id} color='red' height='186px' width='36px' removeFunc={() => removeFunc(id)}/>);
        expect(asFragment).toMatchSnapshot();
    })

})