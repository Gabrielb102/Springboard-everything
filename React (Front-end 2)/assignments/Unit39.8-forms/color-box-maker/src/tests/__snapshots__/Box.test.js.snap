// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for Box Component Renders a consistent user interface 1`] = `[Function]`;
